data/shaders/opengl/basesphere_uniforms.glsl

```diff
@@ -4,16 +4,79 @@
 #include "eclipse.glsl"
 
 layout(std140) uniform BaseSphereData {
-	// to keep distances sane we do a nearer, smaller scam. this is how many times
-	// smaller the geosphere has been made
-	vec3 geosphereCenter;				// TODO documentation
-	float geosphereRadius;				// (planet radius)
-	float geosphereInvRadius;			// 1.0 / (planet radius)
-	float geosphereAtmosTopRad;			// in planet radii
-	float geosphereAtmosFogDensity;		// TODO documentation
-	float geosphereAtmosInvScaleHeight; // TODO documentation
+	// To render accurate planets at any distance or scale, almost all
+	// calculations take place in "planet space", with a distance of 1.0
+	// defined as equivalent to the planet's nominal radius (planet radii).
+	//
+	// This allows planets to be rendered in view-space at any scaling factor
+	// desired, to play nicely with the depth buffer and any other content
+	// rendered (e.g. map views)
+	//
+	// The atmosphere simulation approximates Rayleigh scattering by
+	// calculating the optical density of a path through the atmosphere and
+	// assuming a constant in/out scattering factor based on the density
+	// approximation.
+
+	vec3 geosphereCenter;				// view-space center of the planet, in planet radii
+	float geosphereRadius;				// real planet radius, in meters
+	float geosphereInvRadius;			// 1.0 / (view-space radius), converts between view coordinates and planet radii
+	float geosphereAtmosTopRad;			// height of the simulated atmosphere, in planet radii
+	float geosphereAtmosFogDensity;		// atmospheric density scalar
+	float geosphereAtmosInvScaleHeight; // 1.0 / (atmosphere scale height) in planet radii
 	vec4 atmosColor;
+	vec3 coefficientsR;
+	vec3 coefficientsM;
+	vec2 baricStep;
```

Comment:
These should be described with comments indicating their purpose and their values.